### ISSUE TITLE
[実験] Playing Actor-Critic の Actor / Critic ネットワークを完全分離する

### ISSUE THREAD

**Issue by eletim** [eletim/napoleon-web#153]
## 背景

Playing RL の現行 `actor-critic-v1` は、Actor と Critic が hidden trunk を共有し、最後の `policy_head` / `value_head` のみ分岐する構成になっている。

50/50 opponent pool（current×1、残り4席は各席独立で RuleBased / frozen rl-v740 を 50% / 50% 抽選）で、6000 games / iteration の学習を行ったところ、v0 の評価勝率 65.2% に対して v50 で 65.4% まで戻った後、v100 では 62.4% まで低下した。

一方、full diagnostics では Critic の value loss は概ね改善している。

- iter 9: valueLossBefore ≈ 0.860
- iter 49: ≈ 0.664
- iter 89: ≈ 0.632
- iter 109: ≈ 0.627

Actor 側では同一 self-play dataset に対する `actorLossAfter > actorLossBefore` が多く、shared trunk に流れる Critic gradient が既存の強い policy 表現を損ねている可能性を切り分けたい。

## 目的

Actor と Critic のネットワークを完全に分離した Actor-Critic 構成を追加し、現行 shared-trunk 構成と同条件で比較できるようにする。

## 要件

- Actor と Critic は hidden layer を含めて重みを一切共有しない。
- Actor は既存 Playing policy と同じ MLP 構成・policy logits contract を維持する。
  - 現在の large 構成では `6246 -> 512 -> 512 -> 256 -> 256 -> 53`。
- Critic は独立した MLP とし、同じ observation/model input から scalar value を出力する。
  - 比較実験では Actor と同じ hidden widths を使用してよい。
- 既存 policy checkpoint から Actor-Critic へ移行する場合、Actor 側の初期 logits は元 checkpoint と一致すること。
- Critic は新規初期化する。
- Actor loss の backward で Critic parameter を変更せず、Critic/value loss の backward で Actor parameter を変更しないこと。
- rollout / ONNX export では Actor のみを使用し、Critic を C++ inference 側へ持ち込まない。
- checkpoint / resume / provenance で shared-trunk 型と完全分離型を識別し、誤った resume を fail-close する。
- diagnostics で Actor と Critic の parameter delta を独立に確認できること。
- 既存 shared-trunk `actor-critic-v1` の挙動を壊さない。必要なら新しい algorithm/model architecture ID を追加する。

## 比較実験条件

実装後は、現行 run と条件を揃えて 100〜150 iterations 程度の比較ができること。

- 初期 Actor checkpoint: 現行 50/50 pool run の v0 と同一
- opponent pool: current×1 + 各相手席 RuleBased 50% / frozen rl-v740 50% 独立抽選
- 6000 games / iteration
- batch size 2048
- epochs 1
- Actor learning rate 1e-5
- value loss coefficient 0.5（必要なら内部で Actor/Critic optimizer を分けてもよいが、比較条件として学習率は明示する）
- C++ simulation backend
- rollout concurrency 256
- inference max batch size 256
- rollout / training とも CUDA
- evaluation interval 10
- evaluation seed count 100
- 現行 run と同じ evaluation seeds

## Acceptance criteria

- Actor / Critic が hidden layer を含め完全分離されていることをテストで保証する。
- 既存 policy checkpoint からの migration 直後に Actor logits が一致することをテストする。
- Actor-only / Critic-only update の parameter isolation をテストする。
- ONNX export と C++ rollout が Actor のみで従来どおり動作する。
- checkpoint save/load/resume が完全分離構成で動作し、architecture mismatch は fail-close する。
- 既存 shared-trunk Actor-Critic の regression test が通る。
- 1 iteration の smoke run で C++ rollout -> training -> ONNX export -> C++ evaluation が完走する。

## 非目的

この Issue では以下は変更しない。

- reward 定義
- opponent pool 比率
- PPO / KL clip / entropy bonus 等の新しい RL objective
- learning rate の最適化
- C++ game/simulation rules
- observation/model-input schema

この Issue の目的は、shared trunk による Actor/Critic gradient interference の有無を切り分けること。
